databuilder/query_language.py

```diff
@@ -752,13 +753,18 @@ def decorator(cls):
             raise ValueError("path does not exist")
 
         schema = get_table_schema_from_class(cls)
-        if "patient_id" not in schema:
-            raise ValueError("patient_id must be in table definition")
-        if path.endswith(".csv"):
-            rows = csv.read_dataset_csv(path, schema)
-        elif path.endswith(".csv.gz"):
+        column_specs = {
+            name: ColumnSpec(type=col_type)
+            for name, col_type in dict(
+                [("patient_id", int)] + schema.column_types
+            ).items()
+        }
+
+        if path.suffix == ".csv":
+            rows = csv.read_dataset_csv(path, column_specs)
+        elif path.suffix == ".csv.gz":
```

Comment:
It would have been good to rebase this change into 56d1fe56. I don't understand why we now pass `column_specs` to `read_dataset_csv` when we passed `schema` before.